pages/admin/lessons.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useState } from 'react'
+import { AdminLessonsSideBar } from '../../components/AdminLessonsSideBar'
+import LoadingSpinner from '../../components/LoadingSpinner'
+import Layout from '../../components/Layout'
+import { withGetApp, GetAppProps } from '../../graphql'
+import { Lesson } from '../../graphql/index'
+type AdminLessonsProps = {
+  lessons: Lesson[] | undefined
+  setLessons: React.Dispatch<React.SetStateAction<Lesson[] | null>>
+}
+
+const AdminLessons: React.FC<AdminLessonsProps> = ({ lessons, setLessons }) => {
+  const [selectedLesson, setSelectedLesson] = useState(0)
+  return (
+    <div className="row mt-4">
+      <AdminLessonsSideBar
+        setLessons={setLessons}
+        lessons={lessons}
+        setSelectedLesson={setSelectedLesson}
+      />
+      <h1>{selectedLesson}</h1>
+    </div>
+  )
+}
+
+const Lessons: React.FC<GetAppProps> = ({ data }) => {
+  const { loading, error, lessons, session } = data
+  const [lessonsList, setLessons] = useState<null | Lesson[]>(null)
+  return (
+    <Layout>
+      {loading && <LoadingSpinner />}
+      {error && <h1>Error</h1>}
+      {session && session.user && session.user.isAdmin === 'true' && (
+        <AdminLessons
+          lessons={lessonsList || lessons}
+          setLessons={setLessons}
+        />
+      )}
```

Comment:
instead of doing it like this, it is better to be explicit in your return. Like... there is no scenario where loading is truthy and `session.user` information exist.

```
if (loading) {
  return (
     <Layout> <Loading Spinner /> </Layout>
  )
}
if (error) {
  return (....)
}
const isAdmin = _.get(session, 'user.isAdmin', false)
if (!isAdmin) {
  return (<Layout><h1>You must be admin to access this page</h1></Layout>)
}
return (
...
)
```